src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -620,40 +627,41 @@ def generate_config(
         r['files']['/etc/prometheus/alerting/custom_alerts.yml'] = \
             self.mgr.get_store('services/prometheus/alerting/custom_alerts.yml', '')
 
-        return r, sorted(self.calculate_deps())
+        return r, self.get_dependencies(self.mgr)
 
-    def calculate_deps(self) -> List[str]:
+    @staticmethod
+    def get_dependencies(mgr: "CephadmOrchestrator") -> List[str]:
```

Comment:
@phlogistonjohn TBH I'm not sure it's the case as we don't use any class argument (such as `cls`). This method only relies on the mgr argument in general so I think in this case is in fact a `staticmethod`.

Reply:
@phlogistonjohn 

Yes it does! thank your for your patience and for the detailed explanation :)

In fact, I didn’t look/notice the specific example you pointed out regarding the reference to the class itself for Prometheus in the code.:

`port = cast(int, mgr.get_module_option_ex('prometheus', 'server_port', PrometheusService.DEFAULT_MGR_PROMETHEUS_PORT))`

I think this usage alone is strong enough to justify using a `classmethod` instead of a `staticmethod` in this case. I’ve updated the code to use a `classmethod` in all the methods.